src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.AssemblyReference.cs

```diff
@@ -0,0 +1,230 @@
+                {
+                    return Task.FromResult(SpecializedCollections.SingletonEnumerable<CodeActionOperation>(
+                        new InstallWithPackageManagerCodeActionOperation(reference)));
+                }
+
+                private class InstallWithPackageManagerCodeActionOperation : CodeActionOperation
+                {
+                    private readonly PackageReference reference;
+
+                    public InstallWithPackageManagerCodeActionOperation(PackageReference reference)
+                    {
+                        this.reference = reference;
+                    }
+
+                    public override string Title => FeaturesResources.Install_with_package_manager;
+
+                    public override void Apply(Workspace workspace, CancellationToken cancellationToken)
+                    {
+                        reference._installerService.ShowManagePackagesDialog(reference._packageName);
+                    }
+                }
+            }
+
+            private class InstallNugetPackageOperation : CodeActionOperation
+            {
+                private readonly Document _document;
+                private readonly IPackageInstallerService _installerService;
+                private readonly string _packageName;
+                private readonly string _versionOpt;
+                private readonly List<string> _projectsWithMatchingVersion;
+
+                public InstallNugetPackageOperation(
+                    IPackageInstallerService installerService, Document document, string packageName, string versionOpt)
+                {
+                    _installerService = installerService;
+                    _document = document;
+                    _packageName = packageName;
+                    _versionOpt = versionOpt;
+                    if (versionOpt != null)
+                    {
+                        const int projectsToShow = 5;
+                        var otherProjects = installerService.GetProjectsWithInstalledPackage(
+                            _document.Project.Solution, packageName, versionOpt).ToList();
+                        _projectsWithMatchingVersion = otherProjects.Take(projectsToShow).Select(p => p.Name).ToList();
+                        if (otherProjects.Count > projectsToShow)
+                        {
+                            _projectsWithMatchingVersion.Add("...");
+                        }
+                    }
+                }
+
+                public override string Title => _versionOpt == null
+                    ? string.Format(FeaturesResources.Find_and_install_latest_version_of_0, _packageName)
+                    : string.Format(FeaturesResources.Use_locally_installed_0_version_1_This_version_used_in_2,
+                        _packageName, _versionOpt, string.Join(", ", _projectsWithMatchingVersion));
```

Comment:
It seems like this Title is going to get super long, with 5 project names in it.